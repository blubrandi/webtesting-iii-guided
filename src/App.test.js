import React from 'react';
import banana from 'react-test-renderer'; // 1: install this npm module as a dev dependency

import App from './App';

describe('<App />', () => {
  // 2. write this test

  //Add .only to run only that specific test
  // it.only('matches snapshot', => )
  it('runs the tests', () => {
    expect(true).toBe(true)
  })

  // add .skip to skip this test
  // it.skip('matches snapshot', => )
  it('matches snapshot', () => {
    const tree = banana.create(<App />); // generates a DOM tree

    // snapshots are a JSON representation of the DOM tree
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
