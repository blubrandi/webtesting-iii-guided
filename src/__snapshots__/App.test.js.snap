// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h2>
      Learn React
    </h2>
  </header>
</div>
`;
